listenbrainz/db/follow_list.py

```diff
@@ -0,0 +1,144 @@
+import sqlalchemy
+from listenbrainz import db
+from listenbrainz.db.exceptions import DatabaseException
+from flask import current_app
+
+def _create(connection, name, creator, private=False):
+    result = connection.execute(sqlalchemy.text("""
+        INSERT INTO follow_list (name, creator, private)
+             VALUES (:name, :creator, :private)
+          RETURNING id
+    """), {
+        'name': name,
+        'creator': creator,
+        'private': private,
+    })
+    return result.fetchone()['id']
+
+
+def _add_users(connection, list_id, user_ids):
+    connection.execute(sqlalchemy.text("""
+        INSERT INTO follow_list_member (list_id, user_id, priority)
+             VALUES (:list_id, :user_id, :priority)
+    """), [{'list_id': list_id, 'user_id': user_id, 'priority': priority} for priority, user_id in enumerate(user_ids[::-1])])
+
+
+def _remove_users(connection, list_id):
+    connection.execute(sqlalchemy.text("""
+        DELETE FROM follow_list_member
+              WHERE list_id = :list_id
+    """), {'list_id': list_id})
+
+
+def _get_members(connection, list_id):
+    result = connection.execute(sqlalchemy.text("""
+        SELECT user_id, musicbrainz_id, created, priority
+          FROM follow_list_member
+          JOIN "user"
+            ON follow_list_member.user_id = "user".id
+         WHERE list_id = :list_id
+      ORDER BY priority DESC
+    """), {
+        'list_id': list_id,
+    })
+    return [dict(row) for row in result.fetchall()]
+
+
+def _get_by_creator_and_name(connection, creator, list_name):
+    r = connection.execute(sqlalchemy.text("""
+        SELECT id
+          FROM follow_list
+         WHERE creator = :creator
+           AND LOWER(name) = LOWER(:list_name)
```

Comment:
Is this the way to do this in postgres? Seems to me that that would result in a table scan for each query....

Reply:
I wonder if there is an index trick for this one...

Reply:
There is an index on (creator, list_name) already. I'll look more into it.